src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -54,6 +54,15 @@ public class KestrelServerOptions
         /// </remarks>
         public bool AllowSynchronousIO { get; set; } = false;
 
+        /// <summary>
+        /// Gets or sets a value that controls whether the header values materialized as strings will be reused across requests;
+        /// if they match, or if the strings will always be reallocated.
+        /// </summary>
+        /// <remarks>
+        /// Defaults to true.
+        /// </remarks>
+        public bool ReuseRequestHeaders { get; set; } = true;
```

Comment:
This option is only being provided to disable the feature if something breaks, right? In that case should it be an AppContext switch rather than an API?

Reply:
I was thinking more if your org had some paranoia constraints; since headers contain the cookie and auth headers. However, that should be more unusual than the norm.

How would I set it up as an AppContext switch? Are there examples?

Reply:
https://github.com/aspnet/AspNetCore/wiki/Quirks-mode-switches-in-patches

Reply:
> Quirks mode switches are currently used only in patches. Quirks mode switches are not used in major/minor versions.

Could just get rid of the option?

Reply:
Depends on our confidence. Might do keep the switch for a few previews.